### ISSUE TITLE
require.go Documentation says functions return bool when they don't

### ISSUE THREAD

**Issue by ckeeney** [stretchr/testify#368]
For example, in require.go [require/require.go#L396-L405](https://github.com/stretchr/testify/blob/master/require/require.go#L396-L405) has

```go
// True asserts that the specified value is true.
//
//    assert.True(t, myBool, "myBool should be true")
//
// Returns whether the assertion was successful (true) or not (false).
func True(t TestingT, value bool, msgAndArgs ...interface{}) {
	if !assert.True(t, value, msgAndArgs...) {
		t.FailNow()
	}
}
```

This issue applies to many functions in require.go, and possibly elsewhere.

**Comment by ernesto-jimenez** [comment]
@ckeeney yes, that happens because the code generation tool that outputs that code is copying the documentation from the original assertions. You can see it also uses `assert.True` as an example.

We'll have to figure out how to edit those comments out in the generated code.

Some like [require/require.go#L306](https://github.com/stretchr/testify/blob/master/require/require.go#L306) are even trickier.

**Comment by nmiyake** [comment]
Removing the `Returns` shouldn't be too bad -- most of the comments in assert end in a single line of the form `Returns whether the assertion was successful (true) or not (false).` (and the few that don't can be modified to do this). With this constraint, it should be fairly easy to modify the code generator to take a flag such that, when run for `require`, last lines of this form (either matching exactly or matching a single-line regexp `Returns .+\.` ) in comments are simply trimmed/omitted.

Similarly, it's possible to find indented blocks and replace `assert.` with `require.`.

The case that is much more complicated is the one you cite -- if example code shows using a returned value, there's no easy transform I can think of for that. However, looking at the existing examples, I don't think the example code showing return values really adds much. If we simply modify the example code in those cases, then the rules above would suffice to create correct documentation.

If this sounds okay please let me know and I can open a PR that implements this.

**Comment by brackendawson** [comment]
I'm going to make #1776 the main since it has the most work under it.
